### ISSUE TITLE
False positive in UnnecessarySafeCall (1.11.0-RC2)

### ISSUE THREAD

**Issue by vladimirfx** [detekt/detekt#2961]
## Observed Behavior
Issue reported for necessary safe call:

    val linkedDevices: Map<Device, DeviceLink>
    ...
    linkedDevices[ownerDevice]?.isOwned = true

## Your Environment

* Version of detekt used: 1.11.0-RC2
* Version of Gradle used (if applicable): 6.6
* Operating System and version: MacOS 10.15.5

**Comment by Julesssss** [comment]
Possibly seeing a similar false positive `UnnecessarySafeCall` result in older versions too. 

```
class ExampleViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val textExample: TextView? = itemView.findViewById(R.id.textExample)
}
```

```
// incorrectly flagged as unnecessary
viewHolder.textExample?.typeface = getTypeface()
```

**Comment by schalkms** [comment]
Thanks for listing and tracking further cases here.

**Comment by cortinico** [comment]
Hey @vladimirfx @Julesssss 
I've tested all of your examples removing the types that are not known (or that are Android specifics) but trying to maintain the patterns:

OP Example: https://pl.kotl.in/-VLlrSCsk
Example A: https://pl.kotl.in/ydew3b-Fc
Example B: https://pl.kotl.in/OGo8X_wPe

For all of them, I was not able to reproduce the issue.
Moreover, this specific rule is relying only on an inspection of the Kotlin compiler: `UNNECESSARY_SAFE_CALL`. There is no extra logic added. 

Are you also getting warnings from the Kotlin compiler for those examples?
Can I ask you to try with the snippets I linked if the issue is still reported?

**Comment by Julesssss** [comment]
Sure, I'll try this next week. Thanks for investigating.

**Comment by vladimirfx** [comment]
1. Bug present in type resolution mode (detektMain)
2. Detekt task in the beginning outputs:

    warning: classpath entry points to a non-existent location: /Users/user/.gradle/caches/modules-2/files-2.1/com.github.spotbugs/spotbugs-annotations/4.1.1/f2c06572ee999f374c0849d696305d203bf9c3e/spotbugs-annotations-4.1.1.jar;/Users/user/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/26.0-jre/6a806eff209f36f635f943e16d97491f00f6bfab/guava-26.0-jre.jar;...

all classpath delimeted by ';' (shouldn't it be ':' in Unix system?).

Afterward 10k lines with errors such:

    //Users/vladimirfx/dev/projects/fcb/mfc/mfc-backend/src/main/kotlin/internal/user/webservice/UserProfile.kt:9:18: error: cannot access built-in declaration 'kotlin.String'. Ensure that you have a dependency on the Kotlin standard library
    val surName: String,
                 ^

Detekt 1.10 works correctly in same project. Gradle downgrade is not helps.

**Comment by vladimirfx** [comment]
Looks like bug is there:
https://github.com/detekt/detekt/blob/b75cf4227afca0ca95fcf0ebea426c15a9fd7b7d/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/CliArgument.kt#L42

IMO it should be https://docs.oracle.com/javase/6/docs/api/java/io/File.html#pathSeparatorChar

**Comment by arturbosch** [comment]
> Looks like bug is there:
> https://github.com/detekt/detekt/blob/b75cf4227afca0ca95fcf0ebea426c15a9fd7b7d/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/CliArgument.kt#L42
> 
> IMO it should be:
> 
> ```
> java.io.File.pathSeparatorChar
> ```
> 
> instead of hardcoded ';'

Thanks for investigating this!
This line was not a problem but the handling of the separators in `core`. I've created a PR to fix this regression for the next bugfix release.

**Comment by arturbosch** [comment]
We had a regression in 1.11.0, please update to 1.11.1, thanks!
